### PR TITLE
Fixed Bug db updateing + batch limit

### PR DESCRIPTION
**Pull request by deanott**
fixed batch size to be large and a extra check for apk existance before updateing db

### CODE REVIEW

**Review by AdamSlack — CHANGES_REQUESTED** [review]
Rename downloadAPK to downloader, then we'll talk :haircut_man:

File: pipeline/scraper/downloadAPK.js [hunk 1]
```diff
@@ -87,8 +88,11 @@ async function main() {
                 return;
```
Comment (AdamSlack) [inline pipeline/scraper/downloadAPK.js]: return; ???

File: pipeline/scraper/downloader.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		}
+            } catch (err) {
+                // TODO: Maybe do something else? Destroying process as we have apks that don't exist in db...
+                logger.err('Err when updated the downloaded app', err);
+            }
+        });
+    }
+}
+
+main();
```
Comment (AdamSlack) [inline pipeline/scraper/downloader.js]: EOL missing

File: pipeline/scraper/downloader.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                await downloadApp(app, appSavePath);
+            } catch(err) {
+                logger.warning('Downloading failed with err:', err.message);
+                logger.debug('Attempting to remove created dir');
+                await fs.rmdir(appSavePath).catch(logger.warning);
+                return;
+            }
+
+            try {	
+                console.log(path.join(appSavePath, app.app + '.apk'));
```
Comment (AdamSlack) [inline pipeline/scraper/downloader.js]: Console.log...

File: pipeline/scraper/downloader.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    downloadProcess.stderr.on('data', data => {
+        logger.warning('DL process %d stderr:', downloadProcess.pid,
+            data.toString().replace(/\n/g, util.format('\nDL process %d stderr: ', downloadProcess.pid)));
+    });
+
+    return apkDownloader;
+}
+
+async function main() {
+    for (;;) {
```
Comment (AdamSlack) [inline pipeline/scraper/downloader.js]: Terrible.

File: pipeline/scraper/downloader.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+        await Promise.each(apps, async (app) => {
+            logger.info('Performing download on', app.app);
+
+            try {
+                var appSavePath = await resolveAPKDir(app);
+            } catch (err) {
+                logger.warning('Did not have access to resolve dir', err.message);
+                await new Promise(resolve => setTimeout(resolve, 6000));
+		return;
```
Comment (AdamSlack) [inline pipeline/scraper/downloader.js]: Do you really wanna return nothing?

File: pipeline/scraper/downloader.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            }
+
+            try {
+                await downloadApp(app, appSavePath);
+            } catch(err) {
+                logger.warning('Downloading failed with err:', err.message);
+                logger.debug('Attempting to remove created dir');
+                await fs.rmdir(appSavePath).catch(logger.warning);
+                return;
```
Comment (AdamSlack) [inline pipeline/scraper/downloader.js]: Another ```return;```

Comment (AdamSlack) [reply pipeline/scraper/downloader.js]: Perhaps ```throw(err)``` and catch at the ```promise.each(...)```?

**Review by AdamSlack — APPROVED** [review]

**Comment by deanott** [comment]
Thanks :)

**Pull request merged**